rust/src/conf.rs

```diff
@@ -101,6 +105,42 @@ impl ConfNode {
         return Self { conf }
     }
 
+    pub fn get_child_node(&self, key: &str) -> Option<ConfNode> {
```

Comment:
@jasonish thoughts about this commit ? (wrt to bindgen)

It looks to me that we do not want to have C struct `ConfNode` exported to rust, but instead just have an opaque pointer and some rust-style functions to handle it, so I went this way